eng/common/scripts/Generate-PR-Diff.ps1

```diff
@@ -0,0 +1,51 @@
+<#
+.SYNOPSIS
+Script used to generate the diff.json file for a PR. Explicitly intended to work in a PR context.
+
+.DESCRIPTION
+Combines the result of git diff, some parsed details from the diff, and the PR number into a single JSON file. This JSON file is intended for use further along the pipeline.
+
+.PARAMETER ArtifactPath
+The folder in which the result will be written.
+
+.PARAMETER TargetPath
+The path under which changes will be detected.
+#>
+[CmdletBinding()]
+Param (
+  [Parameter(Mandatory=$True)]
+  [string] $ArtifactPath,
+  [Parameter(Mandatory=$True)]
+  [string] $TargetPath
+)
+
+. (Join-Path $PSScriptRoot "Helpers" git-helpers.ps1)
```

Comment:
orthogonal to this change but at some point I wanted to update these helpers to match what we have done in the specs repo at https://github.com/Azure/azure-rest-api-specs/blob/main/eng/scripts/ChangedFiles-Functions.ps1#L1. It helps us stop depending on the devops variables.

Reply:
Ah ok. I'll submit a PR on that + check its usages in a later PR. Thanks Wes.